roscala/src/main/scala/coop/rchain/rosette/prim/Number.scala

```diff
@@ -84,18 +84,373 @@ object Number {
     @checkTypeMismatch[Fixnum]
     @checkArgumentMismatch
     override def fn(ctxt: Ctxt): Either[PrimError, Fixnum] = {
-      val m = ctxt.argvec.elem.head.asInstanceOf[Fixnum].value
-      val n = ctxt.argvec.elem(1).asInstanceOf[Fixnum].value
+      val m = ctxt.argvec.elem.head.asInstanceOf[Fixnum]
+      val n = ctxt.argvec.elem(1).asInstanceOf[Fixnum]
 
       try {
-        Right(Fixnum(m % n))
+        Right(m % n)
       } catch {
         case e: ArithmeticException =>
           Left(ArithmeticError)
       }
     }
   }
 
+  object fxLt extends Prim {
+    override val name: String = "fx<"
+    override val minArgs: Int = 2
+    override val maxArgs: Int = 2
+
+    @checkTypeMismatch[Fixnum]
```

Comment:
In Rosette `fx<` just checks if the first argument is a `Fixnum`.
If it is and the second argument is not, it returns `RblBool(false)`.

Reply:
Since checkTypeMisMatch is checked at compile time, if I check this in runtime,  is that what you want? how to deal with such case ?

Reply:
`checkTypeMismatch` is expanded to something at compile time which checks types at runtime.
Here you would have to remove `@checkTypeMismatch[Fixnum]` and "manually" check the first argument.